.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseHolidayExtractor.cs

```diff
@@ -30,10 +31,13 @@ public List<ExtractResult> Extract(string text, DateObject reference)
             var ers = Token.MergeAllTokens(tokens, text, ExtractorName);
             foreach (var er in ers)
             {
-                er.Metadata = new Metadata
+                // If this is a daterange that contains a holiday, we should change its
+                // type to indicate that.
+
+                if (!string.IsNullOrEmpty(er.Metadata.HolidayName))
                 {
-                    IsHoliday = true,
```

Comment:
A holiday weekend is still a holiday. Can't we reuse the flow?

Reply:
When the 'IsHoliday' flag is set, it makes the code flow towards the DateTime parsing/resolution (Single day holidays). 

In this instance the name is a bit misleading because what it effectively means is that the extracted entity is entirely a holday, like 'labor's day' or 'thanksgiving'. Should I change it to something like 'IsExactHoliday' or 'IsDateTimeHoliday'?

Reply:
I think so, or use two entries in metadata when relevant.

Reply:
I've went with the latter. Having the additional property "IsHolidayRange" makes this more readable.